### PR TITLE
fix: updating the styles of contribution documentation

### LINKED ISSUE #38589: Update the styles on the contributing documentation
We want to update the styles and theme on the [contributing documentation](https://contribute.freecodecamp.org) site. [Here is a guide](https://contribute.freecodecamp.org/#/how-to-work-on-the-docs-theme) on how to work on the theme and styles.

We would want to:

1. Remove the extraneous CSS 
   - I had shamelessly copied the code from Netflix's awesome pollyjs project [here](https://github.com/Netflix/pollyjs/blob/master/docs/assets/styles.css) while setting up the site.
   - The CSS above depends on the `vue.css` that ships with `docsify`.
2. Write our own theme with [`docsify-themeable`](https://jhildenbiddle.github.io/docsify-themeable/#/).
3. ~Have the Universal Nav~ (already done).
4. Fonts and colors should follow our [design style guide](https://design-style-guide.freecodecamp.org/).
5. Use larger font size similar to our main sites

### PR DESCRIPTION
**Pull request by niteshseram**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `master` branch of freeCodeCamp.
- [x] All the files I changed are in the same world language, for example: only English changes, or only Chinese changes, etc.

Refers #38589

I have made a new custom css to change the font and color. I have also added the feature of Dark and Light Mode.

### CODE REVIEW

**Review by raisedadead — CHANGES_REQUESTED** [review]
This is a great start! Thanks a lot for taking the time for working on it.

I have a few recommendations:

1. Our top nav is always the same in both light and dark modes.
2. The font is a tad overwhelming on the sidebar, we should use the regular Lato there.
3. Could the switch be moved a little more to the left, it seems a bit squished on the right to me.


image


Also, if you have a chance it would be awesome to use a dedicated theme using the `docsify-themeable`

**Comment by raisedadead** [comment]
> Also, if you have a chance it would be awesome to use a dedicated theme using the `docsify-themeable`

This can come in a separate PR.

I am changing the closes keyword to reflect the same from the OP.

**Comment by niteshseram** [comment]
> 1. Our top nav is always the same in both light and dark modes.
> 2. The font is a tad overwhelming on the sidebar, we should use the regular Lato there.
> 3. Could the switch be moved a little more to the left, it seems a bit squished on the right to me.

I can quickly fix 1 and 2. And about 3, I can move it to the left, but since its position is fixed, it comes at the top of the text in mobile view. @raisedadead what do you think, should I remove the fixed  positioning from top? If I removed that, then it solves the problem and I can move it to the left.

**Comment by niteshseram** [comment]
> > Also, if you have a chance it would be awesome to use a dedicated theme using the `docsify-themeable`
> 
> This can come in a separate PR.
> 
> I am changing the closes keyword to reflect the same from the OP.

You mean say to create a custom theme using `docsify themeable` customization? Sorry, I have never used it before so a little confused here. Does it make a difference to use a custom theme or docsify theme?

**Comment by raisedadead** [comment]
> Does it make a difference to use a custom theme or docsify theme?

I have included links to the plugin on the original issue thread. You should give its documentation a read, if you are intrested in implementing it.

Essentially it makes configuring the colors, and other things quite neat vs the hacky way we have it right now. 

The current setup could become a deal breaker long term as we expand on adding more docs.

**Comment by raisedadead** [comment]
> I can quickly fix 1 and 2.

I would recommend checking out the design pattern on the forum. Especially with the 'Dawn theme' (you can select it from the menu).

It has a decent asthetic to the color & typography.



> And about 3, ...

You should check and use the top navbar with the logo. It would be easier to add it there.

**Comment by niteshseram** [comment]
> > Does it make a difference to use a custom theme or docsify theme?
> 
> I have included links to the plugin on the original issue thread. You should give its documentation a read, if you are intrested in implementing it.
> 
> Essentially it makes configuring the colors, and other things quite neat vs the hacky way we have it right now.
> 
> The current setup could become a deal breaker long term as we expand on adding more docs.

@raisedadead  I will go through the documentation. So, I have the changes for the above 3 points you mention. So should I commit the changes for this?

**Comment by raisedadead** [comment]
> So should I commit the changes for this?

I would focus on this PR's needs for the current changes and work on the plugin later on a separate PR after this PR has been merged.

**Comment by niteshseram** [comment]
@raisedadead I have made the changes for the 3 points you mention above. Have a look to it.

File: docs/_theme.css
Comment (raisedadead) [inline docs/_theme.css:44]: Why the need to use a hex code and not a var like everywhere else?

File: docs/_coverpage.md
Comment (raisedadead) [inline docs/_coverpage.md:10]: What is this change? Is it intentional?

**Review by raisedadead — CHANGES_REQUESTED** [review]
In addition to the code comments. I am not convinced with the asthetics. It is off than the current live version.

Lets rethink this a little with feedback from @ahmadabdolsaheb.

Here is some side by side for comparision:

**The muted text color seems to look nicer than the style guide complaint black?**

image

**The legibility is a degrade from the current, should we not use Lato?**

image

**All text is not accounted for in dark mode**

image

**I am not a fan of the slider. This should be a text toggle?? [Light / Dark]**

image

**Illegible.**

image

**Comment by niteshseram** [comment]
@raisedadead I will look into all the points as soon as possible. Meanwhile we can wait for some feedback from @ahmadabdolsaheb.

File: docs/_coverpage.md
Comment (niteshseram) [reply docs/_coverpage.md:10]: Yes, it was. When switch to dark mode, git icon become invisible. So to change the color of git icon.

Comment (ojeytonwilliams) [reply docs/_coverpage.md:10]: We probably should take care to match GitHub's colours.  From using a colour picker on GitHub itself, it looks like they use white and transparent for the light icon and #171515 and transparent for dark.  Not black, but it is close.

**Comment by ahmadabdolsaheb** [comment]
Thank for working on this. Here are some general pointers:
- you could use global CSS variables and switch between them based on light/dark mode. Here is where you could find them.
https://github.com/freeCodeCamp/freeCodeCamp/blob/master/client/src/components/layouts/variables.css

- use the convention from /learn for buttons.
- only use monospace for large titles.
- the nav button should use the same convention used for /learn universal nav. It should be dark/light text and white background on hover. 

Here are some small inconsistencies I noticed.
- the divider between side navigation turns blue on dark mode.
- the "Code of Conduct" text turns white on night mode.

**Comment by ahmadabdolsaheb** [comment]
Hi @niteshseram, I was wondering how the progress is going with this PR?  Please let us know if you have any further questions.

**Comment by niteshseram** [comment]
@ahmadabdolsaheb I am really sorry.I havn't made any progress. I got caught up with my final semester works and examination. But planning to start soon just after my final semester examination which is ending at 5th August.

**Comment by ahmadabdolsaheb** [comment]
Best of luck with your exams @niteshseram.

**Comment by raisedadead** [comment]
The OP has since deleted their fork. We are unable to update the PR. Closing as stale.

**Pull request closed without merging**
